source/particle/world.cc

```diff
@@ -0,0 +1,889 @@
+
+      std::vector<Vector<double> >  values(particles_in_cell,value);
+      std::vector<std::vector<Tensor<1,dim> > > gradients(particles_in_cell,gradient);
+
+      std::vector<Point<dim> >     particle_points(particles_in_cell);
+
+      typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
+      for (unsigned int i = 0; it!=end_particle; ++it,++i)
+        {
+          const Point<dim> position = it->second.get_location();
+          particle_points[i] = this->get_mapping().transform_real_to_unit_cell(cell, position);
+        }
+
+      const std::vector< double > weights(particles_in_cell, 1./((double) particles_in_cell));
+      const Quadrature<dim> quadrature_formula(particle_points, weights);
+      FEValues<dim> fe_value (this->get_mapping(),
+                              this->get_fe(),
+                              quadrature_formula,
+                              update_values |
+                              update_gradients);
+
+      fe_value.reinit (cell);
+      fe_value.get_function_values (this->get_solution(),
+                                    values);
+      fe_value.get_function_gradients (this->get_solution(),
+                                       gradients);
+
+      it = begin_particle;
+      for (unsigned int i = 0; it!=end_particle; ++it,++i)
+        {
+          property_manager->initialize_one_particle(it->second,
+                                                    values[i],
+                                                    gradients[i]);
+        }
+    }
+
+    template <int dim>
+    void
+    World<dim>::local_update_particles(const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                       const typename std::multimap<types::LevelInd, Particle<dim> >::iterator &begin_particle,
+                                       const typename std::multimap<types::LevelInd, Particle<dim> >::iterator &end_particle)
+    {
+      const unsigned int particles_in_cell = std::distance(begin_particle,end_particle);
+      const unsigned int solution_components = this->introspection().n_components;
+
+      Vector<double> value(solution_components);
+      std::vector<Tensor<1,dim> > gradient (solution_components,Tensor<1,dim>());
+
+      std::vector<Vector<double> >  values(particles_in_cell,value);
+      std::vector<std::vector<Tensor<1,dim> > > gradients(particles_in_cell,gradient);
+
+      std::vector<Point<dim> >     particle_points(particles_in_cell);
+
+      typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
+      for (unsigned int i = 0; it!=end_particle; ++it,++i)
+        {
+          const Point<dim> position = it->second.get_location();
+          particle_points[i] = this->get_mapping().transform_real_to_unit_cell(cell, position);
+        }
+
+      const std::vector< double > weights(particles_in_cell, 1./((double) particles_in_cell));
```

Comment:
you don't actually need the weights. Just remove this variable and initialize the Quadrature with only the locations. Same in the function above.